doc/Overview.md

```diff
@@ -19,37 +19,153 @@
+
+This program load a robot model, create a data structure for algorithm buffering, and run the Recursive-Newton-Euler Algorithm (rnea) to compute the robot inverse dynamics.
+
+We first include the proper files. In C++, there is not yet an rationalization of the include files. Here we have to include the sample model (where the parameters of the model we are using in the test are defined), and the header files defining the robot neutral position and the RNEA function. In Python, the library is easier to include by just importing pinocchio.
+
+The first paragraph defines the model. In C++, we first define the object, then allocate it to be our sample model (a human-like kinematic tree with random parameters, very useful for small tests). In Python, the model is created while calling the sample humanoid model.
+The model class contains the constant parameters of the robot (masses, segment length, body names, etc), i.e. the parameters that are not to be modified by the algorithms. As most algorithms need extra room for storage of internal values, we also have to allocate a Data structure, dedicated to the model. Pinocchio relies on a strict separation between constant parameters, in Model, and calculus buffer, in Data.
+
+Inverse dynamics computes the needed torque to track a trajectory defined by position, velocity and acceleration in the configuration space.
+Velocity and acceleration are plain vectors, and can be initialized to any value.
+The configuration might be more complex (e.g. contains quaternion).
+Consequently, helper functions are provides to initialized to zero (neutral) or random configuration, or make sure a configuration is valid (normalize).
+
+Finally, we call the rnea function, by providing the robot model, the corresponding data, and the current configuration, velocity and acceleration.
+The result is a vector of dimension model.nv. It is also stored in data.tau if needed.
+We just print it, also the corresponding values are difficult to interpret.
+
+\section OverviewComplex More complex example with C++ & Python
+
+<table class="manual">
+<tr><th>C++ version</th><th>Python version</th></tr>
+<tr><td>
+\include overview-urdf.cpp
+</td>
+<td>
+\include overview-urdf.py
+</td></tr></table>
+
+\subsection OverviewComplexCompile Compiling and running your program
+
+This time, we must specify that URDFDOM is needed, as the model will be parsed from URDF.
+
+\code g++ -I /usr/include/eigen3/  -I  /path/to/pinocchio/include -L /path/to/pinocchio/lib -DURDFDOM_TYPEDEF_SHARED_PTR -DWITH_URDFDOM  ../examples/sample-urdf.cpp  -lpinocchio -o sample-urdf \endcode
+
+The program typically runs with a UR5 URDF description, that can be found for example in this repository https://github.com/humanoid-path-planner/universal_robot
+
+Launch the program from the directory containing the ur5.urdf file
+\code cd /path/to/ur5.urdf && ./sample-urdf \endcode
+
+In Python, just run it:
+
+\code python sample-urdf \endcode
```

Comment:
I would rename sample-urdf by something like loading-urdf

Reply:
In the example dir, I yet named the 2 tests "overview-*" and then choose arbitrary names in the doc. I guess we should unified, and name the tests overview-simple and overview-urdf .

Reply:
I followed Nicolas' advice